test/test_bpdn.jl

```diff
@@ -19,11 +19,9 @@ function test_bpdn()
     
     # Compute the RHS vector
     b = A * x
-    bl = -ones(n)
-    bu = +ones(n)
     
     # Solve the basis pursuit problem
-    tracer = bpdual(A, b, 0., bl, bu, homotopy = false, loglevel =0)
+    tracer = asp_bpdn(A, b, 0.0, loglevel =0);
 
     xx, λ = tracer[end]
```

Comment:
this seems to expect the solution and lambda, but tracer only stores and returns the solution? I don't understand how this test can pass?

Reply:
no, asp_bpdn uses the bpdual kernel... that does return lambda as well.

Reply:
Sorry I missed this - there are two different tracer types, one returns lambde, the other only the solution. fine then .